### ISSUE TITLE
'strict' rule is only partially overridden by `sourceType: module`

### ISSUE THREAD

**Issue by nre** [eslint/eslint#4936]
v2.0.0-beta.1

The documentation does not cover the behaviour of the 'strict' rule when strict mode is imposed by a parser option: `sourceType: module` (and, hopefully one day, `ecmaFeatures: impliedStrict` #4832).

The behaviour is also inconsistent. The 'strict' option 'none' incorrectly implies that strict mode can be somehow disabled inside ES6 modules ('Strict mode is not permitted'):

```json
{
	"parserOptions": {
		"sourceType": "module"
	}
}
```
```js
/* eslint strict: [2, "never"] */
"use strict";
```
```
  2:1  error  Strict mode is not permitted  strict

✖ 1 problem (1 error, 0 warnings)
```

Whereas the 'strict' options 'global|function|safe' are overridden by the module configuration ("'use strict' is unnecessary inside of modules"):

```json
{
	"parserOptions": {
		"sourceType": "module"
	}
}
```
```js
/* eslint strict: [2, "global"] */
"use strict";
```
```
  2:1  error  'use strict' is unnecessary inside of modules  strict

✖ 1 problem (1 error, 0 warnings)
```

I would like to make a PR that:
- Has ES6 modules override all 'strict' rule options. (This is not a breaking change. Only the message will change.)
- Documents the behaviour of the 'strict' rule and ES6 modules.
- Clarifies the docs to say that the 'strict' rule polices strict-mode *directives* and not strict mode itself. (The 'strict' rule only looks for directives&mdash;it does not use the data in escope. It is a third independent implementation of the strict-directive algorithm, [in addition to the parser and escope](https://github.com/estools/escope/issues/92#issuecomment-168532171).)

**Comment by nzakas** [comment]
What message would change? What would it change to?

**Comment by nre** [comment]
The message that I propose to change is the message that is shown for each strict-mode directive when `sourceType: module` and the 'strict' rule is set to 'never'.

The message would change from **"Strict mode is not permitted"** to **"'use strict' is unnecessary inside of modules"**.

This change would make the behaviour of the 'strict' rule the same for all its options ('never|global|function|safe') when `sourceType: module`.

**Comment by michaelficarra** [comment]
:+1:

**Comment by nzakas** [comment]
Sounds good, let's do it.
